### PR TITLE
[interop][SwiftToCxx] add additional type representation emission che…

### PR DESCRIPTION
**Pull request by hyp**
…ck for associated enum element types

This ensures that we do not try to emit enums whose associated values come from dependent modules that don't have a C++ representation

Also remove outdated FIXME for enum availability.

### CODE REVIEW

**Comment by hyp** [comment]
@swift-ci please test

**Comment by hyp** [comment]
@swift-ci please test

**Comment by hyp** [comment]
@swift-ci please test source compatibility

File: lib/PrintAsClang/DeclAndTypePrinter.cpp [hunk 1]
```diff
@@ -2793,12 +2792,43 @@ static bool isExposedToThisModule(const ModuleDecl &M, const ValueDecl *VD,
...
+  auto *enumDecl = dyn_cast<EnumDecl>(VD);
+  if (!enumDecl)
+    return true;
+  // The supported set of enum elements is restricted by
+  // the types that can be represented in C++. We already
+  // check for different type categories in `getDeclRepresentation`, however,
+  // we also need to perform additional check on whether the type can be
+  // emitted here as well, to ensure that we don't emit types from dependent
+  // modules that do not have a C++ representation.
+  for (const auto *enumCase : enumDecl->getAllCases()) {
```
Comment (ravikandhadai) [inline lib/PrintAsClang/DeclAndTypePrinter.cpp:2806]: Just going by the comment, is there any code duplication between the checks done in `getDeclRepresentation` and the checks here. Is there an opportunity to factor this out to a common helper function?

**Review by ravikandhadai — APPROVED** [review]

**Pull request merged**